enhancements/console/customize-catalogs.md

```diff
@@ -0,0 +1,184 @@
+---
+title: customize-catalogs
+authors:
+  - "@lokanandaprabhu"
+reviewers:
+  - "@jerolimov"
+  - "@spadgett"
+approvers:
+  - "@spadgett"
+  - "@cvogt"
+api-approvers:
+  - "@jerolimov"
+  - "@cvogt"
+creation-date: 2022-07-27
+last-updated: 2022-08-03
+tracking-link: 
+  - https://issues.redhat.com/browse/ODC-6695
+status: implementable
+---
+
+# Customize Catalogs
+
+## Summary
+
+This document provides details about how cluster administrator 
+can hide developer catalog and sub-catalogs. 
+
+## Motivation
+
+Cluster administrator wants to disable developer catalog or sub-catalogs for all users.
+
+### User Stories
+
+#### Story 1
+
+As a cluster admin, I want to hide Developer catalog for all users.
+
+#### Story 2
+
+As a cluster admin, I want to hide sub-catalogs for all users.
+
+### Goals
+
+- Allow cluster admins to hide developer catalog and sub-catalogs.
+- Provide information to admin about Catalogs and sub-catalogs they can hide from users.
+
+### Non-Goals
+
+N/A
+
+## Proposal
+
+We can extend the existing `operator.openshift.io/v1` / `Console CRD`. 
+We can add 
+`spec.customization.catalog.disabled` which can have list of perspectives disabled.
+
+### Workflow Description
+
+Add the disabled catalogs list along with the sub-catalogs to hide. If types are not
+mentioned then complete catalog will be disabled.
+
+```text
+├── Spec
+│   ├── customization
+|       ├── catalog 
+|           ├──disabled
+|             ├── name
+|             ├── types
+|                 └──         
+└── ...
+```
+
+Each catalog conforms to the following schema:
+
+- `disabled?: authV1.ResourceAttributes[]` , define list of catalogs to hide.
+- `name?: authV1.ResourceAttributes` , defines name of catalog to hide.
+- `types?: authV1.ResourceAttributes[]` , defines a list of sub-catalogs to hide.
+
+### API Extensions
+
+Extend the existing `Console` CRD to add `spec.customization.catalog.disabled` which can have list of catalogs and sub-catalogs disabled.
+
+So the final yaml should look like this one:
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: Console
+metadata: 
+  name: cluster
+  ...
+spec: 
+  customization:
+    catalog:
+      disabled:
+        - name: developerCatalog
+          types: 
+            - Devfile
+            - HelmChart
+  ...
+```
+
+To disable the complete Developer Catalog:
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: Console
+metadata: 
+  name: cluster
+  ...
+spec: 
+  customization:
+    catalog:
+      disabled:
+        - name: developerCatalog
+          types: []
```

Comment:
This is a little confusing as an API. I wouldn't expect removing types to disable the entire catalog. Maybe we need an enum value to declare whether we're only disabling certain types or the entire catalog? We should request an API review.